packages/ckeditor5-heading/src/title.ts

```diff
@@ -160,7 +160,7 @@ export default class Title extends Plugin {
 		const editor = this.editor;
 		const data = editor.data;
 		const model = editor.model;
-		const root = editor.model.document.getRoot()!;
+		const root = editor.model.document.getRoot( this.editor.model.document.selection.getFirstRange()!.root.rootName! )!;
```

Comment:
General comment to whole title feature.

You either didn't check or didn't understand what this feature does, as the way it operates cannot operate on current selection state, as it doesn't make any sense. This feature is a bit more complex so it should raise some questions on your side.

Most importantly - if title plugin is used in multi-root editor, should the title be added to each root? I assumed yes, but maybe that's not so obvious. But for now, for simplification, let's assume yes. Later we may add a config value, where the integrator will specify in array root names, for which the title should be enabled.

So, the title plugin should work for all roots. Every root should have it's own title handling. Every root should manage the title (i.e. always keep title block and body block, this is what the postfixer does).

When `getTitle()` and `getBody()` methods are called, in options, `rootName` should be specified (for multi-root) and the returned content should based on that.